ui/src/app/pages/analysis/gke-app-launcher.spec.tsx

```diff
@@ -0,0 +1,88 @@
+import '@testing-library/jest-dom';
+
+import { Router } from 'react-router';
+import { Route } from 'react-router-dom';
+import { createMemoryHistory } from 'history';
+
+import { AppType } from 'generated/fetch';
+
+import { screen } from '@testing-library/dom';
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { GKEAppLauncher } from 'app/pages/analysis/gke-app-launcher';
+import { userAppsStore } from 'app/utils/stores';
+
+const createHistory = (queryParam: string) => {
+  const history = createMemoryHistory({
+    initialEntries: ['/workspaces/:ns/:wsid/gkeApp'],
+  });
+  history.location.search = queryParam;
+  return history;
+};
+
+const createApp = (appType: AppType = AppType.RSTUDIO) => ({
+  googleProject: 'googleProject',
+  appName: 'rstudioApp',
+  appType: appType,
+  proxyUrls: { app: `https://fakeRStudioUrl/` },
+});
+
+const createRoute = (queryParam: string) => (
+  <Router history={createHistory(queryParam)}>
+    <Route path={`/workspaces/:ns/:wsid/gkeApp`}>
+      <GKEAppLauncher hideSpinner={() => {}} showSpinner={() => {}} />
+    </Route>
+  </Router>
+);
+
+const setup = (
+  queryParam: string = 'appType=RStudio',
+  appType: AppType = AppType.RSTUDIO
+) => {
+  userAppsStore.set({
+    updating: false,
+    userApps: [createApp(appType)],
+  });
+
+  return {
+    container: render(createRoute(queryParam)).container,
+    user: userEvent.setup(),
+  };
+};
+
+const setupEmptyUserStore = () => {
+  userAppsStore.set({
+    updating: false,
+    userApps: [],
+  });
+
+  return {
+    container: render(createRoute('appType=RStudio')).container,
+    user: userEvent.setup(),
+  };
+};
+
+it('Should display iframe with rStudio App Url', async () => {
+  setup();
+  // Confirm IFrame exist
+  expect(screen.getByTitle('Gke-App embed')).toBeInTheDocument();
+  // Confirm IFrame source is same as proxyUrl for Rstudio L24
+  expect(screen.getByTitle('Gke-App embed').getAttribute('src')).toBe(
+    'https://fakeRStudioUrl/'
```

Comment:
can you make this a const, maybe "expectedProxyUrl" ?  That way we know what we're looking at without having to go back in the code here